src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -238,6 +251,102 @@ def _on_acquisition_done(self, future):
         scheduled_future = executor.submit(self._export_data, data, None)
         scheduled_future.add_done_callback(self._on_export_data_done)
 
+    def _acquire_at_features(self, _: wx.Event):
+        """Acquire at the features using the current imaging settings."""
+        # store the focuser position
+        self._good_focus_pos = self._tab_data.main.focus.position.value["z"]
+
+        # hide/show/disable some widgets
+        self._panel.gauge_cryosecom_acq.Show()
+        self._panel.btn_cryosecom_acqui_cancel.Show()
+        self._panel.txt_cryosecom_left_time.Show()
+        self._panel.txt_cryosecom_est_time.Hide()
+        self._panel.btn_cryosecom_change_file.Disable()
+        self._panel.btn_acquire_overview.Disable()
+        self._panel.btn_acquire_features.Disable()
+        self._panel.z_stack_chkbox.Disable()
+        self._panel.streams_chk_list.Disable()
+        self._panel.param_Zmin.Enable(False)
+        self._panel.param_Zmax.Enable(False)
+        self._panel.param_Zstep.Enable(False)
+        # disable the streams settings
+        self._tab.streambar_controller.pauseStreams()
+        self._tab.streambar_controller.pause()
+        self._tab_data.main.is_acquiring.value = True
+
+        zlevels: dict = {}
+        if self._zStackActive:
+            self._on_zstack()  # update the zlevels with the current focus position
+            zlevels = self._zlevels
+
+        filename = self._filename.value
+        stage = self._tab_data.main.stage
+        focus = self._tab_data.main.focus
+        features = self._tab_data.main.features.value
+
+        logging.debug(f"Acquiring at features: {features}")
+
+        self._acq_future = acquire_at_features(
+            features=features,
+            stage=stage,
+            focus=focus,
+            streams=self._acquiStreams.value,
+            zlevels=zlevels,
+            filename=filename,
+            settings_obs=self._tab_data.main.settings_obs,
+        )
+
+        # link the acquisition gauge to the acquisition future
+        self._gauge_future_conn = ProgressiveFutureConnector(
+            future=self._acq_future,
+            bar=self._panel.gauge_cryosecom_acq,
+            label=self._panel.txt_cryosecom_left_time,
+            full=False,
+        )
+
+        self._acq_future.add_done_callback(self._on_feature_acquisition_done)
+        self._panel.Layout()
+
+    @call_in_wx_main
+    def _on_feature_acquisition_done(self, future):
+        self._acq_future = None
+        self._gauge_future_conn = None
+        self._tab_data.main.is_acquiring.value = False
+        self._tab_data.main.focus.moveAbs({"z": self._good_focus_pos})
+
+        # try to get the acquisition data
+        try:
+            data, exp = future.result()
+            self._panel.txt_cryosecom_est_time.Show()
+            self._panel.txt_cryosecom_est_time.SetLabel("Finishing acquisition...")
```

Comment:
I think this is a copy of `_on_acquisition_done()`, but here it shouldn't needed. In that other function, the data is not yet saved, and this is done now, which can take a tiny bit of time. Here, the data is already saved, so there is really nothing left to do.

Reply:
removed